core/src/databases/table.rs

```diff
@@ -702,24 +702,40 @@ impl Row {
                 Value::Bool(bool_val)
             } else {
                 // Various datetime formats
-                let dt: Option<DateTime<Utc>> = [
+                let mut dt: Option<DateTime<Utc>> = [
                     // RFC3339
                     DateTime::parse_from_rfc3339(trimmed).map(|dt| dt.into()),
                     // RFC2822
                     DateTime::parse_from_rfc2822(trimmed).map(|dt| dt.into()),
-                    // Google Spreadsheet format
-                    DateTime::parse_from_str(trimmed, "%d-%b-%Y").map(|dt| dt.into()),
                     // SQL
                     DateTime::parse_from_str(trimmed, "%Y-%m-%d %H:%M:%S").map(|dt| dt.into()),
                     // HTTP date
                     DateTime::parse_from_str(trimmed, "%a, %d %b %Y %H:%M:%S GMT")
                         .map(|dt| dt.into()),
+                    // Google Spreadsheet format
+                    NaiveDate::parse_from_str(trimmed, "%d-%b-%Y").map(|d| {
+                        let dt = d.and_hms_opt(0, 0, 0).unwrap();
+                        dt.and_local_timezone(Utc).unwrap()
+                    }),
                     // Date with full month, zero-padded number, full year
-                    DateTime::parse_from_str(trimmed, "%B %d, %Y").map(|dt| dt.into()),
+                    NaiveDate::parse_from_str(trimmed, "%B %d %Y").map(|d| {
+                        let dt = d.and_hms_opt(0, 0, 0).unwrap();
```

Comment:
Default time part to midnight for Date only strings. I'm black belt Rust dates now